UDx-container/entrypoint.sh

```diff
@@ -21,10 +21,122 @@
 # command such as bash, make, or g++ to be executed inside this
 # container. 
 
-dir=$1
-cmd=$2
+VSQL=/opt/vertica/bin/vsql
+vsdk_dir=$1
+vsdk_cmd=$2
 
-echo cd \"$dir\"
-echo sh -c \"$cmd\"
-cd "$dir"
-sh -c "$cmd"
+ADMINTOOLS="${VERTICA_OPT_DIR}/bin/admintools"
+
+# Vertica should be shut down properly
+function shut_down() {
+    echo "Shutting Down"
+    vertica_proper_shutdown
+    echo 'Stopping loop'
+    STOP_LOOP="true"
+}
+
+function vertica_proper_shutdown() {
+    db=$(${ADMINTOOLS} -t show_active_db)
+    case "$db"x in
+        x) 
+            echo "Database not running --- shutting down"
+            ;;
+        *)
+            echo 'Vertica: Closing active sessions'
+            ${VSQL} -c 'SELECT CLOSE_ALL_SESSIONS();'
+            echo 'Vertica: Flushing everything on disk'
+            ${VSQL} -c 'SELECT MAKE_AHM_NOW();'
+            echo 'Vertica: Stopping database'
+            ${ADMINTOOLS} -t stop_db -d $VERTICA_DB_NAME -i
+            ;;
+    esac
+}
+
+function preserve_config() {
+    # unfortunately, admintools doesn't (always) obey symlinks when
+    # manipulating its admintools.conf file, so we have to move the
+    # entire config directory  
+    if [ -f ${VERTICA_DATA_DIR}/config/admintools.conf ]; then
+        # second and subsequent times starting the container
+        # we have an admintools.conf in ${VERTICA_DATA_DIR}
+        echo "config has already been preserved"
+    else
+        # first time through docker-entrypoint.sh we need to move
+        # the config directory to persistent store
+        echo "Moving config directory tree to persistent store"
+        sudo cp --archive ${VERTICA_OPT_DIR}/config ${VERTICA_DATA_DIR}
+        sudo chown -R ${VERTICA_DB_USER} ${VERTICA_DATA_DIR}/config
+    fi
+    # unfortunately, the symlink is in the container image
+    # so we have to renew it each time
+    if [ ! -L ${VERTICA_OPT_DIR}/config ]; then
+        echo "symlink ${VERTICA_OPT_DIR}/config -> ${VERTICA_DATA_DIR}/config"
+        rm -rf ${VERTICA_OPT_DIR}/config
+        ln -s  ${VERTICA_DATA_DIR}/config  ${VERTICA_OPT_DIR}/config
+    fi
+}       
+
+function initialize_vertica_directories() {
+    # We only do this if necessary
+    if [ ! -d ${VERTICA_DATA_DIR}/${VERTICA_DB_NAME} ]; then
+        # first time through --- create db, etc.
+        mkdir -p ${VERTICA_DATA_DIR}/config
+        preserve_config
+        echo 'Creating database'
+
+        ${ADMINTOOLS} -t create_db \
+                      --skip-fs-checks \
+                      -s localhost \
+                      --database=$VERTICA_DB_NAME \
+                      --catalog_path=${VERTICA_DATA_DIR} \
+                      --data_path=${VERTICA_DATA_DIR}
+
+        echo
+        echo 'Loading VMart schema ...'
+        ${VMART_DIR}/${VMART_ETL_SCRIPT}
```

Comment:
Where are these defined?  Is the intention that it could be set in the Dockerfile if the user needs some schema loaded?

Reply:
left over from the one-node-ce, which includes a VMART database.